flashlight/autograd/backend/cuda/RNN.cpp

```diff
@@ -62,6 +92,10 @@ void rnn_backward(
   DropoutDescriptor dropout(drop_prob);
   RNNDescriptor rnn_desc(
       input.type(), hidden_size, num_layers, mode, bidirectional, dropout);
+  if (input.type() == f16) {
```

Comment:
Yes, it should handle all cases. Note that we reinitialize the descriptor in every iteration. As a result, it uses `f32` except if we instruct it otherwise. Nonetheless, it's good to add a case for `f32` to increase the readability. I went ahead and revised the code accordingly.